controllers/ocirepository_controller.go

```diff
@@ -362,6 +364,39 @@ func (r *OCIRepositoryReconciler) reconcileSource(ctx context.Context, obj *sour
 		return sreconcile.ResultEmpty, e
 	}
 
+	// Verify the image
```

Comment:
Let's move all of this under `if !obj.GetArtifact().HasRevision(revision) {` at line 504. This way we avoid spamming the registry for when the artifact download is no-op and when the artifact is invalid.

Reply:
Maybe also move all the verify code into a separate function to make it easier to test and maintain. Similar to how we do it in GitRepository reconciler, refer https://github.com/fluxcd/source-controller/blob/12b200c5167dc38eb631f9ac8e4d8f1273f26079/controllers/gitrepository_controller.go#L865.

Reply:
+1

Reply:
> Let's move all of this under if !obj.GetArtifact().HasRevision(revision) { at line 504. This way we avoid spamming the registry for when the artifact download is no-op and when the artifact is invalid.

Thanks, done, @stefanprodan, but we(w/@dentrax) wanted to make sure the signature is valid before pulling an image with `crane.Pull`, that's why we put this logic there.

> Maybe also move all these verify code into a separate function to make it easier to test and maintain. Similar to how we do it in GitRepository reconcile

that makes a lot of sense, done, thanks @darkowlzz